code/scripting/api/libs/graphics.cpp

```diff
@@ -795,6 +796,28 @@ ADE_FUNC(drawSphere, l_Graphics, "[number Radius = 1.0, vector Position]", "Draw
 	return ADE_RETURN_TRUE;
 }
 
+ADE_FUNC(draw3dLine, l_Graphics, "vector origin, vector destination, [boolean translucent=true, number thickness=1.0, number thicknessEnd=thickness]", "Draws a line from origin to destination. "
+"The line may be translucent or solid. Translucent lines will NOT use the alpha value, instead being more transparent the darker the color is. "
+"The thickness at the start can be different from the thickness at the end, to draw a line that tapers or expands.", nullptr, nullptr)
+{
+	vec3d *v1 = &vmd_zero_vector;
+	vec3d *v2 = &vmd_zero_vector;
+	bool translucent = true;
+	float thickness = 1.0f;
+	float thicknessEnd = -1.0f;
+
+	if (!ade_get_args(L, "oo|bff", l_Vector.GetPtr(&v1), l_Vector.GetPtr(&v2), &translucent, &thickness, &thicknessEnd))
+		return ADE_RETURN_NIL;
+
```

Comment:
I recommend adding a bounds check for `thickness` here, in case someone specifies a negative value for some reason.

Reply:
Zero works on either end, it just tapers into a single pixel point, which IMO doesn't look very good but might be an effect someone wants. 
![image](https://user-images.githubusercontent.com/4577144/171632689-44577c24-a345-4660-8b79-4031976992f1.png)

Doing zero on both values at the same time, however, makes the line invisible. I'll add a check for negative values on either end and another for zero on both ends, then.

Reply:
I also added similar checks in the `batching_add_line` function just in case it eventually gets used by some other feature added in the future